src/Metadata/Driver/TypedPropertiesDriver.php

```diff
@@ -60,9 +60,9 @@ public function __construct(DriverInterface $delegate, ?ParserInterface $typePar
     private function reorderTypes(array $types): array
     {
         uasort($types, static function ($a, $b) {
-            $order = ['null' => 0, 'true' => 1, 'false' => 2, 'bool' => 3, 'int' => 4, 'float' => 5, 'string' => 6];
+            $order = ['null' => 0, 'true' => 1, 'false' => 2, 'bool' => 3, 'int' => 4, 'float' => 5, 'array' => 6, 'string' => 7];
```

Comment:
the more I look at this the more I think that it would make sense to use the order used in the code (by looking at how the union types are declared) rather that having our own way of sorting.

is there a reason for not doing it?

Reply:
I see 2 risks:
- the order might be forced by Code Formatters, so coding style might impact Serializer behavior. Not sure if such linter exists. 
- It might be confusing why `int|string` gives different result than `string|int`. 
Moved into #1576 as an idea for the future.